### PR TITLE
txn: considering the write type processing prewrite errors

### PR DESCRIPTION
**Pull request by cfzjywxk**
Signed-off-by: cfzjywxk <lsswxrxr@163.com>

### What problem does this PR solve?

Issue Number: #10552

Problem Summary:
The jepsen case append failed with incompatible oreder in the inner test environment:
```
2021-07-10 14:36:12,631{GMT}    INFO    [jepsen test runner] jepsen.core: {:perf
 {:latency-graph {:valid? true},
  :rate-graph {:valid? true},
  :valid? true},
 :workload
 {:valid? false,
  :anomaly-types (:incompatible-order),
  :anomalies
  {:incompatible-order ({:key 3, :values [[4 1 2 5 3] [4 1 2 5 6]]})}},
 :valid? false}

Analysis invalid! (ﾉಥ益ಥ）ﾉ ┻━┻
```

In https://github.com/tikv/tikv/pull/10522, we do not return errors if the prewrite keys are already committed, the returned record type needes to be considered processing errors.

### What is changed and how it works?

What's Changed:
Check the write record type.

### Check List 

Tests 

- Unit test

Side effects

### Release note 

```release-note
NONE
```

### CODE REVIEW

File: src/storage/txn/commands/prewrite.rs [hunk 1]
```diff
@@ -444,12 +444,16 @@ impl<K: PrewriteKind> Prewriter<K> {
             reader: &mut SnapshotReader<impl Snapshot>,
             key: &Key,
         ) -> Result<(Vec<std::result::Result<(), StorageError>>, TimeStamp)> {
-            if let TxnCommitRecord::SingleRecord { commit_ts, .. } =
+            if let TxnCommitRecord::SingleRecord { commit_ts, write } =
```
Comment (youjiali1995) [inline src/storage/txn/commands/prewrite.rs:447]: I prefer
```rust
            match reader.get_txn_commit_record(key)? {
                TxnCommitRecord::SingleRecord { commit_ts, write }
                    if write.write_type != WriteType::Rollback =>
                {
                    info!("prewrited transaction has been committed";
                        "start_ts" => reader.start_ts, "commit_ts" => commit_ts);
                    Ok((vec![], commit_ts))
                }
                _ => Err(prewrite_result.unwrap_err().into()),
            }
```

**Review by sticnarf — APPROVED** [review]
LGTM

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**